### DISCUSSION TITLE
Can I run a script in the postinstall phase that visits a webpage on my website that contains ads?

### DISCUSSION THREAD

**Discussion by MatteoBax** [npm/feedback#983] [General]
Hi,
sorry for the stupid question, but I wanted to know if I can run a script in the postinstall phase that visits a page on my website that contains ads in such a way that my package can be monetized.
Am I breaking any npm rules?

Sorry for my bad english, i'm italian.

**Comment by arcanis** [comment]
Postinstalls are intended to build the packages for the local machine, not to emit ads. As for the npm registry in particular:

> These kinds of commercial content generally aren't acceptable:
>
> - README, package.json, or other content displaying advertisements.
>
> - Packages that display ads at runtime, on installation, or at other stages of the software development lifecycle, such as via [npm scripts](https://docs.npmjs.com/misc/scripts). Packages with code that can be used to display ads are fine. Packages that themselves display ads are not.

**Comment by MatteoBax** [comment]
So is it allowed only while using the package, or is it not allowed at all?

**Reply by ljharb** [reply]
Not allowed at all, I’d wager.
